nitro-runtime-manager/src/nitro_runtime.rs

```diff
@@ -39,39 +37,36 @@ impl PlatformRuntime for NitroRuntime {
         init_session_manager()?;
         // generate the csr
         let csr: Vec<u8> = generate_csr()?;
-        // generate the attestation document
-        let mut att_doc: Vec<u8> = {
-            let mut buffer: Vec<u8> = vec![0; NSM_MAX_ATTESTATION_DOC_SIZE];
-            let mut buffer_len: u32 = buffer.len() as u32;
-            let nsm_fd = nsm_lib::nsm_lib_init();
-            if nsm_fd < 0 {
-                return Err(anyhow!(RuntimeManagerError::NsmLibError(nsm_fd)));
-            }
-            let csr_hash = sha256(&csr);
-            let status = unsafe {
-                nsm_lib::nsm_get_attestation_doc(
-                    nsm_fd,                         // fd
-                    csr_hash.as_ptr() as *const u8, // user_data
-                    csr_hash.len() as u32,          // user_data_len
-                    challenge.as_ptr(),             // nonce_data
-                    challenge.len() as u32,         // nonce_len
-                    std::ptr::null() as *const u8,  // pub_key_data
-                    0 as u32,                       // pub_key_len
-                    buffer.as_mut_ptr(),            // att_doc_data
-                    &mut buffer_len,                // att_doc_len
-                )
-            };
-            match status {
-                nsm_api::api::ErrorCode::Success => (),
-                _ => return Err(anyhow!(RuntimeManagerError::NsmErrorCode(status))),
-            }
-            unsafe {
-                buffer.set_len(buffer_len as usize);
-            }
-            buffer.clone()
+
+        let csr_hash = sha256(&csr);
+        let nsm_fd = nsm_api::driver::nsm_init();
+        if nsm_fd < 0 {
+            let e = nsm_api::api::ErrorCode::InternalError;
+            return Err(anyhow!(RuntimeManagerError::NsmErrorCode(e)));
+        }
+
+        let request = nsm_api::api::Request::Attestation{
```

Comment:
I'd like to do the same here, but `Attestation` is probably a bit too generic for that

Reply:
I think I will import `nsm_api::api::{Request, Response}`, this way it will be `Request::Attestation` and it will not clash with `Response::Attestation`